training/src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,137 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+import wolox.training.exceptions.BookIdMismatchException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.Users;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+@RequestMapping("api/books")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private BookRepository bookRepository;
+
+
+    @GetMapping("/users")
+    public Iterable findAll() {
+        return userRepository.findAll();
+    }
+
+    @GetMapping("/users/{id}")
+    public Users findOne(@PathVariable Long id) {
+        return userRepository.findById(id)
+            .orElseThrow(UserNotFoundException::new);
+    }
+
+    private void generateException(String exception) {
+
+        switch (exception) {
+            case "userNotFoundException":
+                try {
+                    throw new UserNotFoundException();
+                } catch (UserNotFoundException e) {
+                    e.printStackTrace();
+                }
+                break;
+            case "bookIdMismatchException": {
+                try {
+                    throw new BookIdMismatchException();
+                } catch (BookIdMismatchException e) {
+                    e.printStackTrace();
+                }
+            }
+            break;
+            case "bookAlreadyOwnedException": {
+                try {
+                    throw new BookAlreadyOwnedException();
+                } catch (BookAlreadyOwnedException e) {
+                    e.printStackTrace();
+                }
+            }
+            break;
+
+            case "bookNotFoundException": {
+                try {
+                    throw new BookNotFoundException();
+                } catch (BookNotFoundException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    @DeleteMapping("/users/{id}")
+    public void delete(@PathVariable Long id) {
+        Optional<Users> userToDelete = userRepository.findById(id);
+        if (userToDelete.isPresent()) {
+            userToDelete.get().getBooks().removeAll(userToDelete.get().getBooks());
+            userRepository.delete(userToDelete.get());
+        } else {
+            this.generateException("userNotFoundException");
+        }
+    }
+
+
+    @PostMapping("/createUser")
+    @ResponseStatus(HttpStatus.CREATED)
+    public Users create(@RequestBody Users users) {
+        Users newUsers = new Users(users.getUsername(), users.getName(), users.getBirthdate());
```

Comment:
It's not necessary to do this. You can save the `users` var directly.
English tip: you are receiving just one user, so the variable should be called `user` (singular) instead of `users` (plural). This would also make the code more understandable.